src/trace_processor/perfetto_sql/intrinsics/functions/utils.h

```diff
@@ -473,6 +473,69 @@ struct RegexpExtract : public sqlite::Function<RegexpExtract> {
   }
 };
 
+struct HexToDec : public sqlite::Function<HexToDec> {
+  static constexpr char kName[] = "HEX_TO_DEC";
+  static constexpr int kArgCount = 1;
+
+  static void Step(sqlite3_context* ctx, int argc, sqlite3_value** argv);
+};
+
+void HexToDec::Step(sqlite3_context* ctx, int, sqlite3_value** argv) {
+  sqlite::Type type = sqlite::value::Type(argv[0]);
+
+  if (type == sqlite::Type::kNull) {
+    return sqlite::result::Null(ctx);
+  }
+  if (type != sqlite::Type::kText) {
+    return sqlite::result::Error(ctx, "HEX_TO_DEC: argument must be text");
+  }
+
+  std::string hex_str(sqlite::value::Text(argv[0]));
```

Comment:
This is creating a ton of copies and allocations. Please use std::string_view and reduce the number of allocations going on here.